### ISSUE TITLE
`line-is-absent` documentation and XSL comment inaccurately describe the condition

### ISSUE THREAD

**Issue by volodya-lombrozo** [objectionary/lints#1198]
Both `line-is-absent.md` and the inline comment in `line-is-absent.xsl` (lines 8-12) describe the lint incorrectly.

- The markdown says it fires when an object with `@base` lacks its own `@line`. Its example doesn't even trigger the real code (no object named `foo` exists to match).
- The XSL comment says it fires when the `@base` target is *not found* — backwards.

**Actual behaviour** (`line-is-absent.xsl:17-24`): fires when `@base` resolves by name to a target that IS found and itself lacks `@line`.

Both need rewriting to match this.

**Comment by volodya-lombrozo** [comment]
We need more examples on which we report or not `line-is-absent`.

**Comment by volodya-lombrozo** [comment]
1. Dangling reference — no object named foo anywhere.

```
<o>
  <o name="bar" base="foo" line="1"/>
</o>
```
Current: silent.

2. Same-scope target, has @line.
```
<o>
  <o name="foo" line="1"/>
  <o name="bar" base="foo" line="2"/>
</o>
```
Current: silent.

3. Same-scope target, missing @line.
```
<o>
  <o name="foo"/>
  <o name="bar" base="foo" line="2"/>
</o>
```
Current: fires — The @line attribute is absent at foo. (This is the intended core case.)

4. Only a different-scope target exists, has @line.
```
<o>
  <o><o name="bar" base="foo" line="1"/></o>
  <o><o name="foo" line="2"/></o>
</o>
```
Current: silent.

5. Only a different-scope target exists, missing @line.
```
<o>
  <o><o name="bar" base="foo" line="1"/></o>
  <o><o name="foo"/></o>
</o>
```
Current: fires, reporting on the unrelated out-of-scope foo. (This is the false positive from #1155.)

6. Same name in both scopes — in-scope one has @line, out-of-scope one doesn't.
```
<o>
  <o>
    <o name="foo" line="1"/>
    <o name="bar" base="foo" line="2"/>
  </o>
  <o><o name="foo"/></o>
</o>
```
Current: fires, on the unrelated out-of-scope foo — even though the actual referenced foo is fine.

7. Same name in both scopes, both missing @line.
```
<o>
  <o>
    <o name="foo"/>
    <o name="bar" base="foo" line="2"/>
  </o>
  <o><o name="foo"/></o>
</o>
```
Current: fires — but which foo gets named in the message depends on document order, not on which one bar actually resolves to.

8. Reference uses a ξ. prefix.
```
<o>
  <o name="foo"/>
  <o name="bar" base="ξ.foo" line="2"/>
</o>
```
Current: silent, always — the name-stripping only removes Φ., not ξ., so the lookup key ("ξ.foo") never matches any real object name, regardless of whether foo has @line.

**Comment by volodya-lombrozo** [comment]
@yegor256 could you help with this, please?

**Comment by h1alexbel** [comment]
@volodya-lombrozo XMIR schema says we can't use `@base="foo"`:

```text
IllegalArgument 2 error(s) in XML document: cvc-pattern-valid: Value 'foo' is not facet-valid with respect to
pattern '∅|\.(ρ|φ|[a-z][^\s.]*|α[0-9]*)|([Φξ])(\.(ρ|φ|[a-z][^\s.]*|α[0-9]*))*' for type 'fqn'.;cvc-attribute.3:
The value 'foo' of attribute 'base' on element 'o' is not valid with respect to its type, 'fqn'.
```

```xml
<o base="foo"/>      <!-- ❌ invalid -->
<o base=".foo"/>     <!-- ✅ relative -->
<o base="ξ.foo"/>    <!-- ✅ this -->
<o base="Φ.foo"/>    <!-- ✅ fullname -->
```

**Comment by yegor256** [comment]
@volodya-lombrozo the `line-is-absent` should simply fire when `@line` attribute is absent. It must be present in _all_ `<o>` elements.

All other cases [mentioned above](https://github.com/objectionary/lints/issues/1198#issuecomment-5175390564) may have their own lints.

**Comment by volodya-lombrozo** [comment]
@h1alexbel would you like to fix it?

**Comment by h1alexbel** [comment]
@volodya-lombrozo yes

**Comment by h1alexbel** [comment]
@volodya-lombrozo all the cases above, except _1. Dangling reference_ will be covered by #1199. Dangling reference lint can be implemented in https://github.com/objectionary/lints/issues/20 WDYT on this?

**Comment by volodya-lombrozo** [comment]
@h1alexbel sure, it's a good plan

**Comment by h1alexbel** [comment]
@volodya-lombrozo can we close this one?

**Comment by volodya-lombrozo** [comment]
@h1alexbel Sure. Thank your four the help!

**Comment by 0crat** [comment]
@h1alexbel Thanks for closing this issue! You've earned +7 points for this: +8 as a basis; -1 for 3 days of delay. Your running score is [+73](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
